agent/api/container/container.go

```diff
@@ -1072,13 +1072,33 @@ func (c *Container) GetLogDriver() string {
 	hostConfig := &dockercontainer.HostConfig{}
 	err := json.Unmarshal([]byte(*c.DockerConfig.HostConfig), hostConfig)
 	if err != nil {
-		seelog.Warnf("Encountered error when trying to get log driver for container %s: %v", err)
+		seelog.Warnf("Encountered error when trying to get log driver for container %s: %v", c.RuntimeID, err)
 		return ""
 	}
 
 	return hostConfig.LogConfig.Type
 }
 
+// GetLogOptions gets the log 'options' map passed into the task definition.
+// see https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_LogConfiguration.html
+func (c *Container) GetLogOptions() map[string]string {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	if c.DockerConfig.HostConfig == nil {
+		return map[string]string{}
+	}
+
+	hostConfig := &dockercontainer.HostConfig{}
+	err := json.Unmarshal([]byte(*c.DockerConfig.HostConfig), hostConfig)
+	if err != nil {
+		seelog.Warnf("Encountered error when trying to get log configuration for container %s: %v", c.RuntimeID, err)
```

Comment:
why not return error here instead of empty map?

Reply:
This is based on the existing implementation of GetLogDriver. I'm guessing this is because we don't want to do anything more drastic than log and record this error, so since we are already logging it here we can safely just return an empty map.

Reply:
We can return map, err

Reply:
Im not sure it's useful to return the error once we've already logged it, what else are we going to do with it?